include/seqan3/argument_parser/detail/format_parse.hpp

```diff
@@ -323,8 +324,10 @@ class format_parse : public format_base
 
         if (auto it = map.find(in); it == map.end())
         {
+            auto keys = map | std::views::keys | seqan3::views::to<std::vector>;
+            std::ranges::sort(keys, [&map] (auto lhs, auto rhs) { return map[lhs] < map[rhs] ;} );
```

Comment:
Why not sort by key? I think we can only sort by value if the value itself is sortable.

I would want to keep the argument_parser free from (external) ranges stuff.

```suggestion
            auto keys_view = map | std::views::keys;
            std::vector<key_type> keys(keys_view.begin(), keys_view.end());
            std::ranges::sort(keys, [&map] (auto key1, auto key2) { return map[key1] < map[key2] ;} );
```

Reply:
Thanks for getting rid of `views::to`, I somehow always forget about the iterator-pair-constructor.

The value must be an enum, so it's always ordered.
I'd prefer sorting by the enum values because they might have some meaningful order regarding the interface (`low`, `average`, `high` might be in that order when sorted by enum, but not when sorted lexicographically).

E.g. in https://github.com/seqan/iGenVar/blob/master/src/variant_detection/method_enums.cpp there are multiple strings that map to one enum, so it would be nice if the string corresponding to an enum are also next to each other.

Reply:
Hmm https://docs.seqan.de/seqan/3-master-user/interfaceseqan3_1_1named__enumeration.html just says:

> A instance of seqan3::enumeration_names<option_type> must exist and be of type std::unordered_map<std::string, option_type>. 

It doesn't state any assumptions about the option_type.

The idea was to also allow stuff like:

```c++
struct my_option1{};
struct my_option2{};
struct my_option3{};
using option_type = std::variant<my_option1, my_option2, my_option3>;
```

And these don't have an order.

Reply:
you're right.

What about using value-order if they are ordered, i.e. `std::totally_ordered<option_type>`.
And otherwise by key?

I'd figure that this will mostly be used for enums

Reply:
Sound good, we can just change the functor the compares depending on the property :)

Reply:
I'll try both in iGenVar and give you a feedback.